flink-libraries/flink-cep/src/test/java/org/apache/flink/cep/nfa/NFAITCase.java

```diff
@@ -1609,7 +1549,7 @@ public boolean filter(Event value) throws Exception {
 
 		compareMaps(resultingPatterns, Lists.<List<Event>>newArrayList(
 				Lists.newArrayList(ConsecutiveData.startEvent, ConsecutiveData.middleEvent1, ConsecutiveData.middleEvent2, ConsecutiveData.end),
-				Lists.newArrayList(ConsecutiveData.startEvent, ConsecutiveData.middleEvent2, ConsecutiveData.middleEvent3, ConsecutiveData.end)
+				Lists.newArrayList(ConsecutiveData.startEvent, ConsecutiveData.middleEvent1, ConsecutiveData.middleEvent3, ConsecutiveData.end)
```

Comment:
Here the expected result should only be the first pattern, because we are in `consecutive` mode.
The problem  is the 

```
if (quantifier.hasProperty(Quantifier.QuantifierProperty.CONSECUTIVE) &&
    !quantifier.hasProperty(Quantifier.QuantifierProperty.EAGER)) {
                    //inner branching and no invalid elements in between
                    return (IterativeCondition<T>) pattern.getCondition();
```
in the `getInnerIgnoreCondition()` of the `NFACompiler`.